File: Competitive Programming 3/Section 1.3.3/573 - The Snail/Python/main.py
def main():
    while True:
        H,U,D,F=map(int,input().split())
        if H == 0:
            return
        pos = 0 
        F = U * F / 100
        day = 0
        while 0<=pos and pos <= H:
            day+=1
            pos += max(0,U) 
            if pos > H:
                break
            pos -= D
            U -= F
        print(("failure" if pos<0 else "success")+" on day "+str(day))
main()

File: Competitive Programming 3/Section 1.3.3/573 - The Snail/C++/main.cpp
#include<bits/stdc++.h>

using namespace std;

double H,D;
double U,F;

int main(){
    ios_base::sync_with_stdio(0);cin.tie(0);
    while(cin>>H>>U>>D>>F,H){
        double pos = 0 ; 
        F = U * F / 100;
        int day=0;
        for(; 0<=pos && pos <= H ;){
            day++;
            pos += max(0.0,U);
            if(pos>H)
                break;
            pos -= D ; 
            U -= F ; 
        }
        cout << (pos<0?"failure":"success") << " on day " << day << "\n";
    }
}